packages/main/src/TabContainer.js

```diff
@@ -929,6 +989,22 @@ class TabContainer extends UI5Element {
 		return focusableTabs;
 	}
 
+	_checkForContent(item) {
+		let hasContent = false;
+		item.content.forEach(content => {
+			if (content.nodeName !== "#text") {
+				hasContent = true;
+			}
+		});
+		return hasContent;
+	}
```

Comment:
This can be a getter inside Tab.js. 

The check itself is too simple. Having text or HTML comments in the markup will break it.

Test it with
```html
<ui5-tab text="text">
   Content
</ui5-tab>

<ui5-tab text="text"><!-- Comment --></ui5-tab>
```

See how the Button checks if it contains only an icon to get an idea how to approach this.